spacy/cli/train.py

```diff
@@ -277,29 +260,26 @@ def evaluate():
         try:
             weighted_score = sum(scores[s] * weights.get(s, 0.0) for s in weights)
         except KeyError as e:
-            raise KeyError(
-                Errors.E983.format(
-                    dict="score_weights", key=str(e), keys=list(scores.keys())
-                )
-            )
-
+            keys = list(scores.keys())
+            err = Errors.E983.format(dict="score_weights", key=str(e), keys=keys)
+            raise KeyError(err)
         scores["speed"] = wps
         return weighted_score, scores
 
     return evaluate
 
 
 def train_while_improving(
-    nlp,
-    optimizer,
+    nlp: Language,
+    optimizer: Optimizer,
     train_data,
     evaluate,
     *,
-    dropout,
-    eval_frequency,
-    accumulate_gradient=1,
-    patience=0,
-    max_steps=0,
+    dropout: float,
+    eval_frequency: int,
+    accumulate_gradient: int = 1,
+    patience: int = 0,
+    max_steps: int = 0,
```

Comment:
I don't think we need any default values here, we should assume we've gotten them from the config